jmclient/jmclient/support.py

```diff
@@ -218,6 +219,26 @@ def cheapest_order_choose(orders, n):
     """
     return orders[0]
 
+def fidelity_bond_weighted_order_choose(orders, n):
+    """
+    choose orders based on fidelity bond for improved sybil resistance
+
+    * with probability `bondless_makers_allowance`: will revert to previous default
+      order choose (random_under_max_order_choose)
+    * with probability `1 - bondless_makers_allowance`: if there are no bond offerings, revert
+      to previous default as above. If there are, choose randomly from those, with weighting
+      being the fidelity bond values.
+    """
+
+    if random.random() < get_bondless_makers_allowance():
```

Comment:
Answered here:

> Some people run makers not for profit but for their own privacy. Therefore not all makers should be required to have bonds, because such privacy-makers are useful to include in coinjoins too. We could have taker allow say, an eighth (12.5%), of their coinjoin peers to be makers without bonds. They can be chosen randomly from the orderbook without any weighting based on fidelity bond values. Of course these are easy to fake by an adversary so they dont contribute much to sybil resistance.

https://gist.github.com/chris-belcher/18ea0e6acdb885a2bfbdee43dcd6b5af/#taker-algorithm-for-choosing-makers

Reply:
Ah, I see. So they will be added to the pool of makers already with an appropriate weight and the fallback mechanism only ensures they have some chance of being picked even if there are a lot of other makers with fidelity bonds? Sounds good.